content/docs/reference-glossary.md

```diff
@@ -120,47 +121,47 @@ class Welcome extends React.Component {
 
-In most cases you should use controlled components.
+ज्यादातर मामलों में आपको नियंत्रित कौम्पोनॅन्ट्स का उपयोग करना चाहिए।
 
 ## [Keys](/docs/lists-and-keys.html) {#keys}
 
-A "key" is a special string attribute you need to include when creating arrays of elements. Keys help React identify which items have changed, are added, or are removed. Keys should be given to the elements inside an array to give the elements a stable identity.
+"keys" एक विशेष string विशेषता है जिसे आपको एलिमेंट्स की सारणी बनाते समय शामिल करना होगा। keys React को पहचानने में मदद करती हैं कि कौन से आइटम बदल गए हैं, जोड़े गए हैं या हटा दिए गए हैं। तत्वों को एक स्थिर पहचान देने के लिए सरणी के अंदर तत्वों को कुंजी दी जानी चाहिए।
 
-Keys only need to be unique among sibling elements in the same array. They don't need to be unique across the whole application or even a single component.
+Keys को केवल एक ही array में सिबलिंग एलिमेंट्स के बीच अद्वितीय होना चाहिए। उन्हें संपूर्ण एप्लिकेशन या किसी एकल कौम्पोनॅन्ट में विशिष्ट होने की आवश्यकता नहीं है।
 
-Don't pass something like `Math.random()` to keys. It is important that keys have a "stable identity" across re-renders so that React can determine when items are added, removed, or re-ordered. Ideally, keys should correspond to unique and stable identifiers coming from your data, such as `post.id`.
+`Math.random()` की तरह keys को कुछ भी पास मत करो। यह महत्वपूर्ण है कि keys में फिर से रेंडर करने वालों की एक "स्थिर पहचान" हो ताकि React यह निर्धारित कर सके कि आइटम कब जोड़े गए, हटाए गए या फिर से ऑर्डर किए गए। आदर्श रूप से, keys को आपके डेटा से आने वाले अद्वितीय और स्थिर पहचानकर्ताओं के अनुरूप होना चाहिए, जैसे कि `post.id`.
 
 ## [Refs](/docs/refs-and-the-dom.html) {#refs}
 
-React supports a special attribute that you can attach to any component. The `ref` attribute can be an object created by [`React.createRef()` function](/docs/react-api.html#reactcreateref) or a callback function, or a string (in legacy API). When the `ref` attribute is a callback function, the function receives the underlying DOM element or class instance (depending on the type of element) as its argument. This allows you to have direct access to the DOM element or component instance.
+React एक विशेष गुण का समर्थन करता है जिसे आप किसी भी कौम्पोनॅन्ट से जोड़ सकते हैं। `ref` गुण [`React.createRef()` फ़ंक्शन](/docs/react-api.html#reactcreateref) या कॉलबैक फ़ंक्शन, या string (लीगेसी API में) द्वारा बनाई गई ऑब्जेक्ट हो सकती है। जब `ref` गुण कॉलबैक फ़ंक्शन होती है, तो फ़ंक्शन अपने तर्क के रूप में अंतर्निहित DOM एलिमेंट या class इंस्टेंस उदाहरण (एलिमेंट के प्रकार के आधार पर) प्राप्त करता है। यह आपको DOM एलिमेंट या कौम्पोनॅन्ट इंस्टेंस तक सीधी पहुँच प्रदान करने की अनुमति देता है।
```

Comment:
Can you revisit this paragraph

Reply:
`इंस्टेंस उदाहरण (एलिमेंट के प्रकार के आधार पर) प्राप्त करता है` => `इंस्टेंस आर्ग्यूमेंट्स (एलिमेंट के प्रकार के आधार पर) के रूप में प्राप्त करता है`
how about this?

Reply:
seems better